libminifi/include/core/ConfigurableComponent.h

```diff
@@ -59,6 +59,18 @@ class ConfigurableComponent {
   template<typename T>
   bool getProperty(const std::string name, T &value) const;
 
+  template<typename T = std::string>
+  std::enable_if_t<std::is_default_constructible<T>::value, std::optional<T>>
+  getProperty(const std::string& property_name) const {
```

Comment:
It would be nicer I agree, I tried https://github.com/apache/nifi-minifi-cpp/pull/1424/commits/6c775d116236c92105c7be38bb284f309c27887c#diff-15b7e52fd974b055c6f71955af2eb1742390e7916dd98c549b9031a9c4b77e9bR62-R63 but it turns out concepts on apple clang are only partially supported (source: https://en.cppreference.com/w/cpp/compiler_support), [here is the failed CI](https://github.com/martinzink/nifi-minifi-cpp/actions/runs/3647371508/jobs/6159525208) maybe with some time we could make it work :man_shrugging: , but I dont have macOS workstation and didnt want to put too much effort into this so I just reverted the change in https://github.com/apache/nifi-minifi-cpp/pull/1424/commits/a1a262a18fbfbdfd6cd0b3c6bf0dd53e94f4af87

Reply:
OK, we'll wait for the next AppleClang release, then.  Thanks for checking.